packages/@cdklabs/cdk-cicd-wrapper/src/resource-providers/CIDefinitionProvider.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as codebuild from 'aws-cdk-lib/aws-codebuild';
+import { GlobalResources, INTEGRATION_PHASES, IResourceProvider, ResourceContext } from '../common';
+
+/**
+ * Provides CodeBuild BuildSpec for the Synth Step
+ */
+export class CIDefinitionProvider implements IResourceProvider {
+  provide(context: ResourceContext): any {
+    const phaseDefinition = context.get(GlobalResources.PHASE)!;
+    const { applicationQualifier, buildSpec } = context.blueprintProps;
+
+    const defaultBuildSpec = codebuild.BuildSpec.fromObject({
+      version: '0.2',
```

Comment:
Is this never going to change, whats the implication to make this generic?

Reply:
The order of the merging allows the users to override these values, with their own buildSpec file. Here, I just ensured backword compatibility for the PR checking CodeBuild. See here: https://github.com/cdklabs/cdk-cicd-wrapper/pull/84/files#diff-6eaa70aa22c4b15cd333f6d274a39871fa142be61b85b8740265893f4f5ac46aL95